### ISSUE TITLE
Overlaping masks not working on custom dataset

### ISSUE THREAD

**Issue by xberesik** [ultralytics/ultralytics#17323]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello, I would like to ask, when I am training the yolov8 for instance segmentation task directly on roboflow, it gives me overlaping segmentation masks as expected. 
![Untitled](https://github.com/user-attachments/assets/af7eb781-6d59-4c11-8ee3-1d1734cf2162)

However when I am training and testing yolov8n-seg model locally I am struggling to get overlapping masks even when I adjusted parameter overlap_mask=True along with iou and conf as I found in other related issues. Please can you help me with this? What else should I change to get overlapping masks? Code sample:
    model = YOLO('yolov8n-seg.pt')
    results = model.train(data=data_path,
                          project=project,
                          name=name,
                          epochs=200,
                          patience=0,  # Disable early stopping
                          batch=6,
                          imgsz=800,
                          overlap_mask=True,
                          boxes=False,
                          save=True,
                          show=True,
                          verbose=False,
                          conf=0.2,
                          iou=0.7,
                          mask_ratio=2) 
   metrics = model.val()
   results1 = model.predict(source='26.jpeg', conf=0.2, iou=0.6, boxes=False, overlap_mask=True)
   Example of tray output mask with wrong holes: 
![image](https://github.com/user-attachments/assets/b2f4d786-09a7-4406-8868-f1611c761bb8)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @xberesik, thank you for reaching out and sharing your query with us on Ultralytics 🚀! This is an automated response, but rest assured, an Ultralytics engineer will also assist you soon.

It's great to see your enthusiasm in using YOLOv8 for instance segmentation. If this is a 🐛 Bug Report, please make sure to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) that helps us debug any concerns with overlapping masks effectively.

Make sure that the parameters you are setting during testing and training locally mirror those from your successful Roboflow configuration. Sometimes discrepancies in input formats or parameters can lead to unexpected results, such as no overlapping masks even if `overlap_mask=True` is set.

Don't forget to explore the [Docs](https://docs.ultralytics.com) for more insights on enhancing your model performance. The [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) section might offer some useful advice on training configuration.

If you are in search of community interaction or require further insights, feel free to join our growing Ultralytics community:
- Chat with us in real-time on [Discord](https://ultralytics.com/discord) 🎧
- For detailed discussions, visit [Discourse](https://community.ultralytics.com)
- For shared learning experiences, join us on [Subreddit](https://reddit.com/r/ultralytics)

## Upgrade

Please ensure your environment is up-to-date with the latest `ultralytics` package along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) setup with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). You can upgrade using:

```bash
pip install -U ultralytics
```

## Environments

Consider testing your setup in one of our verified environments to rule out environment-specific issues:

- **Notebooks** with GPU support: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Set up a **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- Launch with **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- Deploy using **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

Refer to our CI badge below to confirm the operational status of YOLO:

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all YOLO functionalities should be operational as CI tests ensure the reliability of all [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every day and upon each commit.

Stay tuned, and let us know how it goes! 💪

**Comment by Y-T-G** [comment]
`overlap_mask` is True by default. You need to set it to False for each mask to be independent.

**Comment by xberesik** [comment]
Thank you very much! It seems I completely misunderstood the purpose of the overlap_mask parameter. I have one more question: after making a prediction, is it possible to identify which mask is positioned on top of others? In other words, can we determine which object overlaps another?

**Comment by Y-T-G** [comment]
@xberesik You can use the area of the mask to determine that. Probably need some other heuristics too.

**Comment by xberesik** [comment]
I will try that. @Y-T-G, thank you once again for your help!
